crates/solidity/outputs/npm/tests/src/doc-examples/using-queries.ts

```diff
@@ -0,0 +1,171 @@
+import { repoPath } from "../utils/files";
+import path from "node:path";
+import assert from "node:assert";
+import fs from "node:fs/promises";
+
+import { Language } from "@nomicfoundation/slang/language";
+import { RuleKind } from "@nomicfoundation/slang/kinds";
+import { Query } from "@nomicfoundation/slang/query";
+import { RuleNode, TokenNode } from "@nomicfoundation/slang/cst";
+
+async function parseDocInputFile(filePath: string) {
+  const inputPath = repoPath(path.join("documentation/public/user-guide/inputs", filePath));
+  const source = await fs.readFile(inputPath, "utf8").then((source) => source.trim());
+  const language = new Language("0.8.0");
+  return language.parse(RuleKind.SourceUnit, source);
+}
+
+test("using queries", async () => {
+  {
+    let parse_output = await parseDocInputFile("using-the-cursor.sol");
+    // --8<-- [start:creating-a-query]
+
+    // Any `Cursor` can be used to create a query.
+    let cursor = parse_output.createTreeCursor();
+
+    let query = Query.parse("[ContractDefinition]");
+    let result /* : query.QueryResultIterator */ = cursor.query([query]);
```

Comment:
nit: I suggest uncommenting it to use type checking, and also using `results` instead of `result`, as it is an iterator of multiple ones.

```suggestion
    let results: query.QueryResultIterator = cursor.query([query]);
```